pkg/proxy/iptables/proxier.go

```diff
@@ -254,7 +256,7 @@ func NewProxier(ipt utiliptables.Interface,
 	healthzServer healthcheck.ProxierHealthUpdater,
 	nodePortAddresses []string,
 ) (*Proxier, error) {
-	if utilproxy.ContainsIPv4Loopback(nodePortAddresses) {
+	if disableLocalhostNodePorts != nil && !*disableLocalhostNodePorts && utilproxy.ContainsIPv4Loopback(nodePortAddresses) {
```

Comment:
@danwinship Thank your point!  Maybe like this ？
```golang
if !disableLocalhostNodePorts && utilproxy.ContainsIPv4Loopback(nodePortAddresses) {
		// Set the route_localnet sysctl we need for exposing NodePorts on loopback addresses
		klog.InfoS("Setting route_localnet=1, use nodePortAddresses to filter loopback addresses for NodePorts to skip it https://issues.k8s.io/90259")
		if err := utilproxy.EnsureSysctl(sysctl, sysctlRouteLocalnet, 1); err != nil {
			return nil, err
		}
		for _, chain := range iptablesJumpChains {
			if chain.dstChain == kubeNodePortsChain {
				chain.extraArgs = append(chain.extraArgs,[]string{"!", "-s", "127.0.0.0/8"}...)
			}
		}
}
```

Reply:
Note that this is IPv4-specific.  You need to make it conditional and handle IPv6 ("::1/128")

Reply:
Oh, right, although we should block nodeports to `::1` _regardless of_ the flag, because they never work and always cause kernel warnings.

Rather than modifying `iptablesJumpChains` here, it would probably be cleaner to just remove `kubeNodePortsChain` from that and have it set up specially in `syncProxyRules` after the `iptablesJumpChains` loop

Reply:
thanks for the advice！

PR Updated.  II'm not sure where it would be better to set the `iptables` rule that disables access to `nodePort` service via localhost, I inserted the `iptables` rule in the chain of  `KUBE-NODEPORTS` and table of `nat`, is this appropriate?

Thanks again!

Reply:
> Oh, right, although we should block nodeports to `::1` _regardless of_ the flag, because they never work and always cause kernel warnings.

@danwinship nodeports on localhost works in ipv6 and without route_localnet if the ip origin is not localhost, the security risk AFAIK was that enabling route_localnet  _Do not consider loopback addresses as martian source or destination while routing_

What should we block packets to localhost? is there any additional security issue with that?

Reply:
So for ipv6, Should we block packets whose source and destination are both localhost? Like the following：

```shell
-A KUBE-SERVICES ! -s -d ::1/128  ! -d ::1/128 -m comment --comment "kubernetes service nodeports; NOTE: this must be the last rule in this chain" -m addrtype --dst-type LOCAL -j KUBE-NODEPORTS
```

Reply:
I thought the security risk was eliminated by the anti-martian-packet rule in kubelet?

At any rate, this PR is about "disable nodeport on localhost", not "stop setting `route_localnet`". If the user says to disable nodeports on localhost, then we should make sure nodeports on localhost don't work, regardless of whether they are a security risk or not.

(If this PR had been intended to be "remove insecure parts of the nodeport feature while preserving as much of the other existing functionality as possibility" rather than "disable nodeports on localhost" then it needs some rethinking, and definitely a better option name.)

Reply:
I see, so the goal is to remove nodeports listening on localhost, then we should update the release notes ... this topic